amplify/function/BusinessLogic/ResetHackathon/handler.ts

```diff
@@ -0,0 +1,267 @@
+import { Amplify } from "aws-amplify";
+import { generateClient } from "aws-amplify/data";
+
+import type { Schema } from "../../../data/resource";
+import type { ScoreComponentTypeInput } from "./graphql/API";
+import {
+  deleteScore,
+  deleteTeam,
+  deleteTeamRoom,
+  deleteUser,
+  updateHackathon,
+} from "./graphql/mutations";
+import {
+  listHackathons,
+  listScores,
+  listTeamRooms,
+  listTeams,
+  listUsers,
+} from "./graphql/queries";
+
+Amplify.configure(
+  {
+    API: {
+      GraphQL: {
+        endpoint: process.env.AMPLIFY_DATA_GRAPHQL_ENDPOINT as string,
+        region: process.env.AWS_REGION,
+        defaultAuthMode: "iam",
+      },
+    },
+  },
+  {
+    Auth: {
+      credentialsProvider: {
+        getCredentialsAndIdentityId: async () => ({
+          credentials: {
+            accessKeyId: process.env.AWS_ACCESS_KEY_ID as string,
+            secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY as string,
+            sessionToken: process.env.AWS_SESSION_TOKEN as string,
+          },
+        }),
+        clearCredentialsAndIdentityId: () => {
+          /* noop */
+        },
+      },
+    },
+  },
+);
+
+const client = generateClient<Schema>({
+  authMode: "iam",
+});
+
+type Handler = Schema["ResetHackathon"]["functionHandler"];
+
+export const handler: Handler = async (event) => {
+  try {
+    // Extract arguments from the event if any
+    const {
+      safetyCheck,
+      resetRooms,
+      resetScores,
+      resetTeams,
+      resetUsers,
+      startDate,
+      endDate,
+      scoreComponents,
+      scoringSidepots,
+    } = event.arguments;
+    if (safetyCheck !== process.env.EDIT_HACKATHON_CODE) {
+      return {
+        statusCode: 403,
+        headers: { "Content-Type": "application/json" },
+      };
+    }
+    // get the current hackathon model data items
+    const HackathonItems = (
+      await client.graphql({
+        query: listHackathons,
+      })
+    ).data.listHackathons.items;
+
+    // Assuming there is only 1 Hackathon
+    if (HackathonItems.length === 0) {
+      console.log("Not a single Hackathon, please create one");
+      return {
+        statusCode: 400,
+        headers: { "Content-Type": "application/json" },
+      };
+    }
+    // Assuming the first Hackathon is the right one
+    const HackathonID = HackathonItems[0].id;
+
+    // Resetting Users variable
+    const resettingUsers: boolean = resetUsers ?? false;
```

Comment:
After these event arguments are marked as required as per other comment, we can remove all the ??. It's crucial to remove uncertainty here as these are destructive actions. 

Additional Context and why we should use AppSync to check: Defaulting to false may not always be what we expect. In a situation like this, we want the user to definitely provide true or false and nothing else. Marking them as required in the data.resource file will ensure that AppSync does this strict checking for arguments for us and tells the user something is wrong way before we even make it to this code.